### PR TITLE
Remove RPM prefix from type names

### LINKED ISSUE #158: RPMBuilder and similar types should be named as RpmBuilder
Per [Rust API guidelines](https://rust-lang.github.io/api-guidelines/naming.html):
>In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn. In snake_case, acronyms and contractions are lower-cased: is_xid_start.

Alternatively, we can simply drop the `RPM` prefix and use the types like `rpm::Builder`.

### PR DESCRIPTION
**Pull request by newpavlov**
Closes #158

### CODE REVIEW

File: README.md
Comment (dralley) [inline README.md:37]: I really don't like the lone `Builder` name.  I'm kind of conflicted on the others too, but this stands out as not being a clear name.

Likewise importing `rpm::Error` would be mutually exclusive with `dyn Error`.

My personal preference is to leave the prefix, but I'm fine with changing the casing i.e. `RpmError` and `RpmBuilder` and `RpmPackage`.

@drahnr @cmeister2 Opinions?

Comment (drahnr) [reply README.md:37]: I'd leave the prefix for `Package` and `Builder` at least

Comment (newpavlov) [reply README.md:37]: >I really don't like the lone Builder name.

How about `PackageBuilder`?

>Likewise importing rpm::Error would be mutually exclusive with dyn Error.

In my experience, it's quite common for crates to define an `Error` type. For example, see the [`regex`](https://docs.rs/regex) crate. In some cases, crates even have per-module `Error` types, similarly to `std` which has a whole bunch of them.

>I'd leave the prefix for Package and Builder at least

Honestly, it's a bit weird to see `rpm::RpmPackage` and `rpm::RpmBuilder` in code.

Comment (newpavlov) [reply README.md]: I have renamed `rpm::Builder` to `rpm::PackageBuilder`. I believe `rpm::Package` is a descriptive enough name.

**Review by dralley — APPROVED** [review]
I think I can live with this.  @drahnr @cmeister2 both of you can weigh in before merging

**Comment by dralley** [comment]
Squash the commits into one please

**Comment by dralley** [comment]
@drahnr @cmeister2 do either of you have objections to this?

**Review by cmeister2 — COMMENTED** [review]
Think a lot of these make sense. I could take or leave the Error class rename because having multiple Errors makes things a bit hard to manage, and I'd probably just rename it RpmError on import anyway.

**Comment by newpavlov** [comment]
@cmeister2 
Personally, in the case of potential name conflicts, I usually use `crate_or_mod_name::Error` or `crate_or_mod_name::Result` (if the latter is available), i.e. in this case it would be `rpm::Error`, which is as clear as `RpmError` and does not require any imports.

**Comment by cmeister2** [comment]
Sure, ok. Don't want to bikeshed something trivial. I'll let @drahnr approve.

**Comment by dralley** [comment]
@newpavlov Could you rebase and squash the commits back down to one

**Comment by newpavlov** [comment]
Not right now. It will be faster for you to use the "squash and merge" option, than for me to do it manually.

**Review by dralley — APPROVED** [review]

**Comment by dralley** [comment]
I can't squash and merge because of the conflicts, but I also can't directly perform edits on this PR because it's on your fork.

**Comment by newpavlov** [comment]
>I can't squash and merge because of the conflicts

There are no conflicts, since I've already resolved them in web interface. You should be able to squash and merge this PR without any issues (I've done so many times in repositories which I maintain).

>I also can't directly perform edits on this PR because it's on your fork

You can. The option "Allow edits by maintainers" is enabled by default and I haven't disabled it. You should be able to directly commit into my fork.

**Comment by dralley** [comment]
@drahnr I'm going to preliminarily merge this, it will very likely be sitting for another week or two before any releases happen so if there are objections we can handle them.

**Pull request merged**
